sw/tools/parrot/bebop.py

```diff
@@ -122,16 +122,22 @@ def bebop_status():
     # Split filename and path
     f = parrot_utils.split_into_path_and_file(args.file)
 
-    print("Kill running " + f[1] + " and make folder " + args.folder)
-    parrot_utils.execute_command(tn,"killall -9 " + f[1])
-    sleep(1)
-    parrot_utils.execute_command(tn, "mkdir -p /data/ftp/" + args.folder)
-    print('Uploading \'' + f[1] + "\' from " + f[0] + " to " + args.folder)
-    parrot_utils.uploadfile(ftp, args.folder + "/" + f[1], file(args.file, "rb"))
-    sleep(0.5)
-    parrot_utils.execute_command(tn, "chmod 777 /data/ftp/" + args.folder + "/" + f[1])
-    parrot_utils.execute_command(tn, "/data/ftp/" + args.folder + "/" + f[1] + " > /dev/null 2>&1 &")
-    print("#pragma message: Upload and Start of ap.elf to Bebop succesful !")
+    #check firmware version
+    v = parrot_utils.check_version(tn, '').strip()
+    print("Checking Bebop firmware version... " + v )
+    if v != '3.2.0':
+        print("Error: firmware version on bebop is incorrect!")
```

Comment:
maybe also change that to something like "Error: please upgrade your Bebop firmware to version 3.2.0"?